consensus/core/src/block_verifier.rs

```diff
@@ -80,36 +91,49 @@ impl BlockVerifier for SignedBlockVerifier {
                 committee.size(),
             ));
         }
-        let mut own_ancestor = false;
-        let mut seen_parents = vec![false; committee.size()];
+        if block.ancestors().is_empty() {
+            return Err(ConsensusError::InsufficientParentStakes {
+                parent_stakes: 0,
+                quorum: committee.quorum_threshold(),
+            });
+        }
+        let mut seen_ancestors = vec![false; committee.size()];
         let mut parent_stakes = 0;
-        for ancestor in block.ancestors() {
+        for (i, ancestor) in block.ancestors().iter().enumerate() {
             if !committee.is_valid_index(ancestor.author) {
                 return Err(ConsensusError::InvalidAuthorityIndex {
                     index: ancestor.author,
                     max: committee.size() - 1,
                 });
             }
-            if ancestor.author == block.author() {
-                own_ancestor = true;
+            if (i == 0 && ancestor.author != block.author())
```

Comment:
ooc why is it important that the block.author has its ancestor as the first ancestor in the list?

Reply:
Each block is required to include an own ancestor. It seems good to have a convention on where it is placed. In future we may run more validations on the consistency between a block and its own ancestor. Instead of a separate field, own ancestor is stored at the 1st element of the ancestors Vec.